opta/opta_run_check.py

```diff
@@ -0,0 +1,41 @@
+"""
+Usage: This file is used to check if Opta is already running in the current directory.
+This creates a file in opta directory with the name: .opta_running_<underscored-cwd>_<pid>
+"""
+
+
+import glob
+import os
+
+from opta.constants import HOME
+from opta.exceptions import UserErrors
+
+OPTA_RUNNING_LOC_PREFIX = os.path.join(
+    HOME, ".opta", f".opta_running_{os.getcwd().replace(os.path.sep, '_')}_"
+)
+
+
+def check_opta_running_file_exists() -> bool:
+    return glob.glob(OPTA_RUNNING_LOC_PREFIX + "*").__len__() > 0
+
+
+def create_opta_running_file() -> None:
+    with open(f"{OPTA_RUNNING_LOC_PREFIX}_{os.getpid()}", "w") as f:
+        f.write("")
+
+
+def remove_opta_running_file() -> None:
+    try:
+        os.remove(f"{OPTA_RUNNING_LOC_PREFIX}_{os.getpid()}")
+    except FileNotFoundError:
+        pass
+
+
+def opta_run_start() -> None:
+    if check_opta_running_file_exists():
+        raise UserErrors("Opta already running in the current directory.")
```

Comment:
I haven't checked but only apply/destroy should be included in this check.
we can have a user running `opta apply` and on another terminal doing an `opta logs` to troubleshoot what's going on live, right?

Reply:
The reason for this was that the graceful exits are a part of `cli` and not `apply` and destroy`.
So, if there is by any chance an ungraceful exit, the user would be stuck in a state, making user's job difficult for using opta.

Reply:
Check if this fits the use case: https://github.com/run-x/opta/pull/749/commits/6536250dcde5cab5d713657272a3798806d8a88c